Building MATLAB toolbox throws Python version "Default"

## Description
When building Matlab toolbox, throws 
```
CMake Error at cmake/GtsamMatlabWrap.cmake:218 (find_package):
  find_package called with invalid argument "Default"
Call Stack (most recent call first):
  cmake/GtsamMatlabWrap.cmake:68 (wrap_library_internal)
  gtsam/CMakeLists.txt:220 (wrap_and_install_library)
```

## Steps to reproduce

```
mkdir -p build && cd build
cmake -DGTSAM_INSTALL_MATLAB_TOOLBOX=ON .. 
```

## Expected behavior
Shouldn't be throwing error.

## Environment
- Ubuntu 20.04
- gcc version 9.3.0 (Ubuntu 9.3.0-17ubuntu1~20.04) 
- cmake version 3.16.3
- `python -V` outputs Python 3.8.5
- branch `develop`
## Additional information
None

--- a/cmake/HandlePython.cmake
+++ b/cmake/HandlePython.cmake
@@ -1,4 +1,5 @@
-if(GTSAM_BUILD_PYTHON)
+# Set Python version if either Python or MATLAB wrapper is requested.
+if(GTSAM_BUILD_PYTHON OR GTSAM_INSTALL_MATLAB_TOOLBOX)
     if(${GTSAM_PYTHON_VERSION} STREQUAL "Default")
         # Get info about the Python3 interpreter
         # https://cmake.org/cmake/help/latest/module/FindPython3.html#module:FindPython3
@@ -14,7 +15,9 @@ if(GTSAM_BUILD_PYTHON)
                 "The version of Python to build the wrappers against."
                 FORCE)
     endif()
+endif()
 
+if(GTSAM_BUILD_PYTHON)
     if(GTSAM_UNSTABLE_BUILD_PYTHON)
         if (NOT GTSAM_BUILD_UNSTABLE)
             message(WARNING "GTSAM_UNSTABLE_BUILD_PYTHON requires the unstable module to be enabled.")

